Automattic/jetpack
# Instagram Gallery Block: Add the UI

See Automattic/explorers-private#36

#### Changes proposed in this Pull Request:

* Add a way to connect a widget to an Instagram account and to load a gallery.
* Add the whole UI to control the gallery (post count, columns, padding).

<img width="1405" alt="Screenshot 2020-03-25 at 17 33 37" src="https://user-images.githubusercontent.com/2070010/77567521-2e21d700-6ebf-11ea-9649-549a3d0fa20b.png">

<img width="1407" alt="Screenshot 2020-03-25 at 17 33 26" src="https://user-images.githubusercontent.com/2070010/77567516-2b26e680-6ebf-11ea-9359-8571ff0243e0.png">

#### Notes

- When changing the number of posts displayed, sometimes the request returns an incorrect number of posts. Might be a case of needing to debounce the calls, or something more complicated.
- Even though the gallery fetch is only supposed to be called when the access token or the number of posts change, it's also called when: changing the block width to full or to default, when toggling between code and visual editor. This is unnecessary, since we already have those images saved in the post content.
- The disconnection doesn't have any loading UI. I haven't found a satisfying place for it, but I don't think it's a blocker for testing the PR.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* pb5gDS-nX-p2

#### Testing instructions:






* Use a site connected to WPCOM (JP Docker is ok when tunneled through ngrok).
* Open the editor and add an Instagram Gallery block.
* Connect it to an Instagram account (log into it _before_ attempting to connect from the block).
* Poke around with the block settings and make sure all works as expected both in the editor and in the front end.
* Disconnect the account, and make sure the block reverts to a placeholder state.
* Also make sure that the block content is now empty: check in the code editor, and in the front end.

#### Proposed changelog entry for your changes:

* N/A
